### DISCUSSION TITLE
How do I configure ugs-platform-app to enable my plugin by default?

### DISCUSSION THREAD

**Discussion by vannoo67** [winder/Universal-G-Code-Sender#2607] [Help] [answered]
I have written a version of ugs-platform-plugin-jog (called ugs-platform-plugin-jog4l) to support jogging for 4 linear axis machines, such as foam cutters. I'm not ready to make a Pull Request yet, but I'm close. If you're interested, have a look at https://github.com/vannoo67/Universal-G-Code-Sender/tree/jog4l

I would like to make a distribution of UGS that has jog4l installed and enabled (rather than the original jog). I have figured out how to incorporate the dependency in ugs-platform-app/pom.xml so that the plugin appears as one of the installed plugins, but it is disabled.

```
<!--
        <dependency>
            <groupId>${project.groupId}</groupId>
            <artifactId>ugs-platform-plugin-jog</artifactId>
            <version>${project.version}</version>
        </dependency> -->
        <dependency>
            <groupId>${project.groupId}</groupId>
            <artifactId>ugs-platform-plugin-jog4l</artifactId>
            <version>${project.version}</version>
        </dependency>  
```

How do I configure the project so that jog4l is enabled by default?

**Comment by breiler** [comment]
I think that you will need to add it here as well to get it to be loaded by default:
https://github.com/vannoo67/Universal-G-Code-Sender/blob/jog4l/ugs-platform/application/pom.xml#L83

**Reply by vannoo67** [reply]
Sorry, I typo-ed, that is the file that I have updated. The plugin is indeed included (loaded?) but not enabled.

**Reply by vannoo67** [reply]
By that I mean, the plugin appears in the Plugins | Installed tab, but not activated. I can manually activate it here, but I'd like for it to be active by default.

**Reply by breiler** [reply, marked as answer]
I think the problem is because it has the same name as the original jog component.
Change the name of the component at these locations:
https://github.com/vannoo67/Universal-G-Code-Sender/blob/jog4l/ugs-platform/ugs-platform-plugin-jog4l/src/main/java/com/willwinder/ugs/nbp/jog4l/JogTopComponent.java#L52
https://github.com/vannoo67/Universal-G-Code-Sender/blob/jog4l/ugs-platform/ugs-platform-plugin-jog4l/src/main/java/com/willwinder/ugs/nbp/jog4l/JogTopComponent.java#L65

It will now open by default for me:
![image](https://github.com/user-attachments/assets/e48c66b2-d635-4826-b377-2e6487ef46e9)

**Reply by vannoo67** [reply]
Fantastic. Thankyou so much.

Another question, if I may. I couldn't figure out where to change the title of the plugin tab "Jog Controller". I search/replaced every instance of "Jog Controller" I could find. I would like for the User to be able to distinguish easily between the two if they have both jo & jog4l active.

**Reply by breiler** [reply]
This is done when the component is opened in order to allow localizations:
https://github.com/vannoo67/Universal-G-Code-Sender/blob/jog4l/ugs-platform/ugs-platform-plugin-jog4l/src/main/java/com/willwinder/ugs/nbp/jog4l/JogTopComponent.java#L122

**Reply by vannoo67** [reply]
Thanks for this, much appreciated. I'm curious that the value is populated by a different Localization system than the one that populates the JogPanel strings. For those I created a clone of the ugs-core Localization object, pointing at my local resource folder. I have added the following to it;
```
platform.plugin.jog4l.JogControlTitle = Jog 4L Controller
platform.plugin.jog4l.JogControlTooltip = Manually Control the machine location.
```
Is there any reason that this is a Bad Idea?

**Reply by breiler** [reply]
The motivation for having all localization in the same central file was to allow our translation system (POEditor) to manage all of them throughout the application. This is however not ideal as your plugin will no longer be isolated.

If you plan to have this merged as a part of UGS I'd recommend you to use the central localization for now. I have been looking at another system which allows multiple translation files.

**Reply by vannoo67** [reply]
Thanks for this. I'd love to have jog4l merged into UGS, so I'll do the Localization as you suggest.
